src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -822,6 +823,12 @@ export default class Fms {
                 const fixName = extractFixnameFromHoldSegment(segment);
 
                 isValid = this._navigationLibrary.hasFix(fixName);
+            } else if (RouteModel.isVectorRouteString(segment)) {
+                const heading = extractHeadingFromVectorSegment(segment);
+                const isValidNumber = !isNan(heading);
+                const isThreeDigits = heading.length === 3;
+
+                isValid = isValidNumber && isThreeDigits;
```

Comment:
`<glimpse-into-nates-head>`
I'm not sure this abstraction is needed.  don't change it, though, because it is easy to read.  I look at this and read the logic bits just as I would the consts that get set.  Your head is in the right place here, and this isn't wrong (quite the opposite), but a situation like this I feel is borderline.
`</glimpse-into-nates-head>`